lib/services/sorting/post_sorting_service.dart

```diff
@@ -0,0 +1,59 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:collection/collection.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/post/post_firestore.dart";
+import "package:proxima/services/sorting/post_sort_option.dart";
+
+class PostSortingService {
+  /// This method sorts the [posts] according to the [option]
+  /// and the [position] of the user. It also allows to put some
+  /// posts on top with the [putOnTop] set.
+  /// For instance, let's say that we have a list [A, b, c, D, e]
+  /// randomly shuffled that we sort by alphabetical order. If
+  /// putOnTop contains [A, D, K], then the reuslt will be
+  /// [A, D, b, c, e].
```

Comment:
Well seen for the typo, fixed in 0f0e6f26c6958427a4f3032b1d8ae705846e37ef.

No `K` is not a typo, if an element is in `putOnTop` but not in the list, it will not be added to the list. This is for challenges later so that, if they are in the feed, they appear on top. I improved the documentation in c1909e7af60f5f2a37443cf91e365646da5052cb. Tell me if this is clearer.